openvino_xai/explainer/explanation.py

```diff
@@ -107,21 +113,22 @@ def _select_target_saliency_maps(
         assert self.layout == Layout.MULTIPLE_MAPS_PER_IMAGE_GRAY
         explain_target_indices = self._select_target_indices(
             targets=targets,
-            total_num_targets=len(self._saliency_map),
             label_names=label_names,
         )
         saliency_maps_selected = {i: self._saliency_map[i] for i in explain_target_indices}
         return saliency_maps_selected
 
-    @staticmethod
     def _select_target_indices(
+        self,
         targets: np.ndarray | List[int | str],
-        total_num_targets: int,
         label_names: List[str] | None = None,
     ) -> List[int] | np.ndarray:
         explain_target_indices = get_explain_target_indices(targets, label_names)
-        if not all(0 <= target_index <= (total_num_targets - 1) for target_index in explain_target_indices):
-            raise ValueError(f"All targets explanation indices have to be in range 0..{total_num_targets - 1}.")
+        if self.total_num_targets is not None:
+            if not all(0 <= target_index <= (self.total_num_targets - 1) for target_index in explain_target_indices):
+                raise ValueError(
```

Comment:
This check is applied then self.saliency_map is numpy. Do we need the same check for the case when it's a dict? Like:

```
        if self.total_num_targets is not None: # numpy case
            if not all(0 <= target_index <= (self.total_num_targets - 1) for target_index in explain_target_indices):
                ...
        elif not all(self.saliency_map.get(target_index, False) for target_index in explain_target_indices): # dict case
           ...raise valueerror
```